### PR TITLE
Extend DETR to support tfrecord datasets

### PR DESCRIPTION
**Pull request by gunho1123**
# Description

> :memo: Please include a summary of the change. 
> Current DETR implementation supports TFDS only as an input source. To utilize my dataset or downloaded tfrecord files in the framework, I extend it to support tfrecord inputs on local or cloud storage.
> In addition, I add a backbone factory in modeling to make flexible choosing backbone for feature encoding.
> I have organized the configuration list, which currently contains all the parameters in task configuration. I add subgroups for the parameters to make it clear. (task.model, task.losses, ...)
> No additional dependencies are required.
> TPU compatibility checked.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [O] New feature (non-breaking change which adds functionality)

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**:

## Checklist

- [O] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [O] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [O] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [O] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [O] I have commented my code, particularly in hard-to-understand areas.
- [O] I have made corresponding changes to the documentation.
- [O] My changes generate no new warnings.
- [O] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/projects/detr/tasks/detection_test.py [hunk 1]
```diff
@@ -112,5 +116,127 @@ def test_validation_step(self):
       state = task.aggregate_logs(step_outputs=logs)
       task.reduce_aggregated_logs(state)
 
+class DetectionTest_tfds(tf.test.TestCase):
```
Comment (frederick0329) [inline official/projects/detr/tasks/detection_test.py:119]: DetectionTFDSTest

File: official/projects/detr/tasks/detection_test.py [hunk 2]
```diff
@@ -112,5 +116,127 @@ def test_validation_step(self):
...
+      task = detection.DectectionTask(config)
+      model = task.build_model()
+      metrics = task.build_metrics(training=False)
+      dataset = task.build_inputs(config.validation_data)
+      iterator = iter(dataset)
+      logs = task.validation_step(next(iterator), model, metrics)
+      state = task.aggregate_logs(step_outputs=logs)
+      task.reduce_aggregated_logs(state)
+
+class DetectionTest_tfrecord(tf.test.TestCase):
```
Comment (frederick0329) [inline official/projects/detr/tasks/detection_test.py:180]: DetectionTFRecordTest

Comment (gunho1123) [reply official/projects/detr/tasks/detection_test.py]: Thank you for your suggestion.

**Review by frederick0329 — APPROVED** [review]

File: official/projects/detr/tasks/detection_test.py (hunk 2 above)
Comment (arashwan) [reply official/projects/detr/tasks/detection_test.py]: @gunho1123 thanks for the pr, it sounds like the branch is out of sync, you need to sync to master and resubmit this PR.

**Pull request closed without merging**